src/ast/seq_decl_plugin.cpp

```diff
@@ -1595,3 +1576,108 @@ std::ostream& seq_util::re::pp::display(std::ostream& out) const {
         // Else: derivative or is_of_pred
         return out << mk_pp(e, re.m);
 }
+
+/*
+  Returns true iff info has been computed for the regex r
+*/
+bool seq_util::re::has_valid_info(expr* r) {
+    return r->get_id() < m_infos.size() && m_infos[r->get_id()].is_valid();
+}
+
+/*
+  Get the information value associated with the regular expression e
+*/
+seq_util::re::info seq_util::re::get_info(expr* e) 
+{
+    SASSERT(u.is_re(e));
+    if (!has_valid_info(e)) {
+        m_info_pinned.push_back(e);
+        compute_info_rec(e);
+    }
+    return m_infos[e->get_id()];
+}
+
+/*
+  Computes the info value for the given regex e recursively over the structure of e
+*/
+void seq_util::re::compute_info_rec(expr* e) {
```

Comment:
thus with return value a seq_util::re::info you can use the result in the combiner.